docs/routers/ubiquiti-usg.md

```diff
@@ -0,0 +1,62 @@
+        inet6 2001:470:5:3::8ed  prefixlen 128  scopeid 0x0<global>
+        inet6 2001:470:5:3:8d70:e92e:7030:6eef  prefixlen 64  scopeid 0x0<global>
+        inet6 fe80::337f:710d:6781:4c85  prefixlen 64  scopeid 0x20<link>
+        ether dc:a6:32:0e:31:2d  txqueuelen 1000  (Ethernet)
+        RX packets 1297459  bytes 1279521260 (1.1 GiB)
+        RX errors 0  dropped 0  overruns 0  frame 0
+        TX packets 2315988  bytes 2766370095 (2.5 GiB)
+        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
+```
+
+Take note of the `inet` address, and the `inet6` address which starts with `fe` (denotes a local link or LAN address). Those are the IPs which you'll use as your DNS servers in the USG. For this example, we'll be using the `192.168.0.9` and `fe80::337f:710d:6781:4c85` IPs for the following directions
+
+### Distribute Pi-hole as DNS server via DHCP
+
+Go to **Settings** --> **Networks** and click on the Network line that you want to modify:
+
+![Screenshot of USG LAN](../images/usg-lan.png)
+
+Under **DHCP**, in the DHCP Server Management section click "Show options" to reveal the **DHCP DNS Server** section. Check the **Enable** box and enter your Raspi's IPv4 address here.
+
+![Screenshot of USG LAN DHCP settings](../images/usg-lan-dhcp.png)
+
+Scroll down to **IPv6**, and under the **DHCPv6/RDNSS DNS Control** section, uncheck **Auto** if it's enabled and enter Raspi's IPv6 address here. Click **Apply Changes** to have the settings persistently saved.
+
+![Screenshot of USG LAN DHCPv6 settings](../images/usg-lan-dhcpv6.png)
+
+All client devices on your network should now automatically be configured with a single IPv4 and single IPv6 address for DNS resolution in the IP stack. If you don't see your client devices update quickly enough, you *can* toggle Wi-Fi off/on or disconnect the Ethernet cable for a couple of seconds and then reconnect it.
+
+### Optional Step - Using your Pi-hole as a resolve for your USG
+
+Completing this *optional* step, Pi-hole can be used by your USG as an upstream DNS server, with its DNS queries taking the following path:
+
+```
+(Clients) -> USG -> Pi-hole -> Upstream DNS Server
+```
+
+You may want to do this for a variety of use cases. For example, you may want to have exclude a development or other test environment excluded from your normal IP stack.
+
+you prefer any DNS queries coming directly from your USG to use Pi-hole. In my case, I wanted to enable this so I'm certain 100% of my network's DNS queries were forwarded using [DNS-over-HTTPS](https://docs.pi-hole.net/guides/dns/cloudflared/).
+
+!!! warning
+    Keep in mind that if you are using your USG's IP address as a resolver for your Pi-hole, ***this will cause a resolver loop***. Make sure to only have true upstream DNS servers set in your Pi-hole's configuration, or use Pi-hole as [your own recursive DNS server with Unbound](https://docs.pi-hole.net/guides/dns/unbound/?h=unbound#configure-unbound).
```

Comment:
```suggestion
    Keep in mind that if you are using your USG's IP address as an upstream resolver for your Pi-hole together with this configuration, ***this will cause a resolver loop***. Make sure you have an other upstream DNS servers set in your Pi-hole's configuration, or use Pi-hole as [your own recursive DNS server with Unbound](https://docs.pi-hole.net/guides/dns/unbound/?h=unbound#configure-unbound).
```